File: 0053-maximum-subarray/0053-maximum-subarray.go
func maxSubArray(nums []int) int {

	size := len(nums)

	maxSum := nums[0]
	currentSum := 0

	for i := 0; i < size; i++ {
		currentSum += nums[i]
		if currentSum > maxSum {
			maxSum = currentSum
		}
		if currentSum < 0 {
			currentSum = 0
		}
	}

	return maxSum

}

File: 0053-maximum-subarray/0053-maximum-subarray.java
class Solution {
    public int maxSubArray(int[] nums) {
        int sum = nums[0];
        int currSum = sum;
        for(int i=1; i<nums.length; i++) {
            if(currSum < 0) {
                currSum = 0;
            }
            currSum += nums[i];
            if(currSum > sum) {
                sum = currSum;
            }
        }
        return sum;
    }
}